src/external/listenForDocumentQueries.ts

```diff
@@ -0,0 +1,37 @@
+import { queryApi } from '../client';
+import { FinchQueryOptions } from '../types';
+import { createResponseEvent } from './createEvents';
+import { FinchRequestEvent, FinchDocumentEventNames } from './types';
+
+/**
+ * listenForDocumentQueries binds to document events so websites can
+ * send Finch GraphQL queries and mutations from document events. This
+ * depends if you are able to inject into those pages as well.
+ *
+ * This method will also attach a **data-finch-listener** attribute on
+ * the body element to be able to known the event bound.
+ *
+ * @param options the finch query options that are passed to queryApi method
+ * @returns A function that stops listening to the events
+ */
+export const listenForDocumentQueries = (options?: FinchQueryOptions) => {
+  const onMessage = async (event: FinchRequestEvent) => {
+    const resp = await queryApi(
+      event.detail.query,
+      event.detail.variables,
+      options,
+    );
+    const responseEvent = createResponseEvent(
+      event.detail.requestId,
+      resp.data,
+      resp.errors,
+    );
+    document.dispatchEvent(responseEvent);
+  };
+  document.addEventListener(FinchDocumentEventNames.Request, onMessage);
+  document.body.setAttribute('data-finch-listener', `${Date.now()}`);
```

Comment:
I wonder since this is being added to the body if there is a way to support more than one extension using this type of messaging. Eg. should we do something like `data-finch-<EXTENSION_ID>` so that way we know who is listening. If we change the value to the id then we might need to support an array of ids... Leaning towards the former.

Reply:
I guess this does not really matter, since the listener filters out requests.